src/app/qgisapp.cpp

```diff
@@ -630,6 +630,7 @@ QgisApp::QgisApp( QSplashScreen *splash, bool restorePlugins, bool skipVersionCh
   }
 
   sInstance = this;
+  Qgis::ogrRuntimeSupport();
```

Comment:
If you move the check to somewhere around main.cpp line 531 and just return from the main function it should work.